precompiles/vesting/types.go

```diff
@@ -36,6 +38,32 @@ type Period struct {
 	Amount []cmn.Coin
 }
 
+// CheckApprovalArgs checks the arguments passed to the approve function as well as
+// the functions to change the allowance. This is refactored into one function as
+// they all take in the same arguments.
+func checkApprovalArgs(args []interface{}) (common.Address, string, error) {
+	if len(args) != 2 {
+		return common.Address{}, "", fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 3, len(args))
+	}
+	grantee, ok := args[0].(common.Address)
+	if !ok || grantee == (common.Address{}) {
+		return common.Address{}, "", fmt.Errorf(authorization.ErrInvalidGranter, args[0])
+	}
+
+	typeURL, ok := args[1].(string)
+	if !ok {
+		return common.Address{}, "", fmt.Errorf(authorization.ErrInvalidMethod, args[2])
+	}
+
+	if typeURL == "" {
+		return common.Address{}, "", fmt.Errorf(authorization.ErrInvalidMethod, typeURL)
+	}
+
+	// TODO: check if the typeURLs are valid? e.g. with a regex pattern?
```

Comment:
Just after the call of this function in `approve.go` there is the type check so I think it would be redundant here.